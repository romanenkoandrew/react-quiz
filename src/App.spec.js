import React from "react";
import { shallow } from "enzyme";
import { App } from "./App";
import configureStore from "redux-mock-store";

const mockStore = configureStore();

describe("App component", () => {
  let autoLogin, component, instance;
  beforeEach(() => {
    autoLogin = jest.fn();
    component = shallow(<App autoLogin={autoLogin} />);
    instance = component.instance();
  });
  it("App component without authenticated", () => {
    expect(component).toMatchSnapshot();
  });
  it("App component with authenticated", () => {
    const component2 = shallow(<App autoLogin={autoLogin} isAuthenticated />);
    expect(component2).toMatchSnapshot();
  });
  it("should call autologin", () => {
    expect(autoLogin).toBeCalledTimes(1);
  });
});
