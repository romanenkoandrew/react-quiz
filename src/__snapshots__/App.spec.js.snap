// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component App component with authenticated 1`] = `
<Connect(Layout)>
  <Switch>
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "type": [Function],
        }
      }
      path="/quiz-creator"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "type": [Function],
        }
      }
      path="/quiz/:id"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "type": [Function],
        }
      }
      path="/logout"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "type": [Function],
        }
      }
      exact={true}
      path="/"
    />
    <Redirect
      to="/"
    />
  </Switch>
</Connect(Layout)>
`;

exports[`App component App component without authenticated 1`] = `
<Connect(Layout)>
  <Switch>
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "type": [Function],
        }
      }
      path="/auth"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "type": [Function],
        }
      }
      path="/quiz/:id"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "type": [Function],
        }
      }
      path="/"
    />
    <Redirect
      to="/"
    />
  </Switch>
</Connect(Layout)>
`;
